src/rebootstrap-raft/main.go

```diff
@@ -71,42 +72,59 @@ func (c *rebootstrapCommand) Info() *cmd.Info {
 // SetFlags is part of cmd.Command.
 func (c *rebootstrapCommand) SetFlags(f *gnuflag.FlagSet) {
 	c.CommandBase.SetFlags(f)
-	f.BoolVar(&c.verbose, "verbose", false, "show debug logging")
 	f.BoolVar(&c.dryRun, "dry-run", false, "build the configuration but don't bootstrap raft")
-	f.StringVar(&c.raftDir, "raft-dir", "/var/lib/juju/raft", "raft directory location")
-	f.StringVar(&c.machineID, "machine-id", "", "ID of this Juju controller machine")
+	f.BoolVar(&c.ssl, "ssl", true, "use SSL to connect to MongoDB ")
+	f.BoolVar(&c.verbose, "verbose", false, "show debug logging")
 	f.IntVar(&c.apiPort, "api-port", 17070, "the API port of the Juju controller")
 	f.StringVar(&c.hostname, "hostname", "localhost", "the hostname of the Juju MongoDB server")
+	f.StringVar(&c.jujuDir, "juju-dir", "/var/lib/juju", "path of jujud files")
+	f.StringVar(&c.machineID, "machine-id", "", "ID of this Juju controller machine")
 	f.StringVar(&c.mongoPort, "mongo-port", "37017", "the port of the Juju MongoDB server")
-	f.BoolVar(&c.ssl, "ssl", true, "use SSL to connect to MongoDB ")
 	f.StringVar(&c.password, "password", "", "password for connecting to MongoDB")
 }
 
 // Init is part of cmd.Command.
 func (c *rebootstrapCommand) Init(args []string) error {
 	if c.verbose || c.dryRun {
 		logger.SetLogLevel(loggo.DEBUG)
+
+	}
+	_, c.asSnap = os.LookupEnv("SNAP")
+	if c.asSnap {
+		logger.Debugf("Running inside a snap")
 	}
 	return c.CommandBase.Init(args)
 }
 
+// When we are running inside a confined snap /var/lib gets remounted to /var/lib/hostfs/var/lib.
+// This should be transparent to the user.
```

Comment:
I would also explain the "snapify" boolean.
// If you pass true for snapify, the function will return the redirected path so that you can access the content.
// However, pass snapify=false when reporting paths to the user, since they won't expect the paths to be mounted at a subdirectory.